cdap-ui/app/cdap/services/datasource/index.js

```diff
@@ -115,16 +122,34 @@ export default class Datasource {
     });
     this.eventEmitter.on(WINDOW_ON_FOCUS, this.resumePoll.bind(this));
     this.eventEmitter.on(WINDOW_ON_BLUR, this.pausePoll.bind(this));
+      SystemDelayStore.dispatch({
+        type: SystemDelayActions.registerDataSource,
+        payload: this,
+      });
+  }
+
+  getBindingsForHealthCheck() {
+    const bindingsWithTime={};
+     Object.values(this.bindings)
+        .filter(binding => !binding.excludeFromHealthCheck)
```

Comment:
you can filter this as you are computing whether a binding is delayed or not, don't need to pre filter it

Reply:
I was filtering it here to keep the response simple. I removed the extra `filter` and checking inside the forEach so we loop only once.